web/src/user_groups.ts

```diff
@@ -285,6 +285,15 @@ export function get_realm_user_groups_for_dropdown_list_widget(
         });
 
     if (require_system_group) {
+        if (setting_name === "direct_message_permission_group") {
+            system_user_groups = system_user_groups.map((group) => {
+                if (group.name === "Nobody") {
+                    group.name = "Direct messages disabled";
+                }
+                group.name = group.name.replace(" and ", " or ");
+                return group;
+            });
```

Comment:
I guess this code can be moved above where we set the name parameter. Also, I guess `require_system_group` is `False` for this setting. We can actually do something like have a function like `get_display_name_for_system_group_option` and then have a if-condition for this setting to return these names.